### ISSUE TITLE
Composer throw deprecation for psr-0/psr-4 standard

### ISSUE THREAD

**Issue by rvitaliy** [magento/magento2#28033]
### Preconditions (*)
PHP: 7.3.17
Composer version: 1.10.5
"magento/product-enterprise-edition": "2.3.4"

### Steps to reproduce (*)
run this commands
1. composer install --no-interaction --no-progress --no-suggest --no-dev
2. composer dump-autoload --optimize --no-dev --classmap-authoritative

### Expected result (*)
no depreciation shown

### Actual result (*)
```
$ composer dump-autoload --optimize --no-dev --classmap-authoritative
Generating optimized autoload files (authoritative)
Deprecation Notice: Class Zend_Validate_Barcode_IntelligentMail located in ./vendor/magento/zendframework1/library/Zend/Validate/Barcode/Intelligentmail.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///usr/local/bin/composer/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Yotpo\Yotpo\Cron\UpdateMetadataJob located in ./vendor/yotpo/magento2-module-yotpo-reviews/Cron/UpdateMetadata.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///usr/local/bin/composer/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class Temando\Shipping\Block\Attribute\Mapping\Product located in ./vendor/temando/module-shipping-m2/Block/Adminhtml/Attribute/Mapping/Product.php does not comply with psr-4 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///usr/local/bin/composer/src/Composer/Autoload/ClassMapGenerator.php:201
```

**Comment by rvitaliy** [comment]
same deprecation in "magento/product-enterprise-edition": "2.3.5-p1"

**Comment by hostep** [comment]
For the `Zend_Validate_Barcode_IntelligentMail` error, this issue was created: https://github.com/magento/zf1/issues/28, and a fix is pending: https://github.com/magento/zf1/pull/29 (would be great if this could get picked up one day btw ...)

**Comment by rvitaliy** [comment]
please, fix this issue.
composer v2.0 is coming soon https://github.com/composer/composer/releases/tag/2.0.0-RC2

**Comment by rvitaliy** [comment]
update with magento EE v2.4.1
```
+ composer dump-autoload --optimize --no-dev --classmap-authoritative
Generating optimized autoload files (authoritative)
Deprecation Notice: Class Zend_Validate_Barcode_IntelligentMail located in ./vendor/magento/zendframework1/library/Zend/Validate/Barcode/Intelligentmail.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///usr/local/bin/composer/src/Composer/Autoload/ClassMapGenerator.php:201
Deprecation Notice: Class CBOR\CBORIncorrectAdditionalInfoException located in ./vendor/2tvenom/cborencode/src/CBOR/CBORExceptions.php does not comply with psr-0 autoloading standard. It will not autoload anymore in Composer v2.0. in phar:///usr/local/bin/composer/src/Composer/Autoload/ClassMapGenerator.php:201
Generated optimized autoload files (authoritative) containing 28642 classes
```

**Comment by rvitaliy** [comment]
related to https://github.com/magento/community-features/issues/302

**Comment by hostep** [comment]
@rvitaliy, it looks like these issues are fixed in Magento 2.4.2 and its dependencies, can you confirm?
